packages/hatchify-core/src/assembler.ts

```diff
@@ -0,0 +1,115 @@
+import { integer } from "./dataTypes"
+import type {
+  FinalAttribute,
+  FinalNumberORM,
+  PartialAttribute,
+  PartialNumberControlType,
+  PartialNumberORM,
+} from "./types"
+
+export interface PartialSchema {
+  name: string
+  id?: PartialAttribute<
+    PartialNumberORM,
+    PartialNumberControlType,
+    number,
+    FinalNumberORM
+  >
+  attributes: {
+    [attributeName: string]: PartialAttribute<
+      PartialNumberORM,
+      PartialNumberControlType,
+      number,
+      FinalNumberORM
+    >
+  }
+}
+export interface FinalSchema {
+  name: string
+  id: FinalAttribute<
+    PartialNumberORM,
+    PartialNumberControlType,
+    number,
+    FinalNumberORM
+  >
+  attributes: {
+    [attributeName: string]: FinalAttribute<
+      PartialNumberORM,
+      PartialNumberControlType,
+      number,
+      FinalNumberORM
+    >
+  }
+}
+
+type PartialSchemaWithPrimaryAttribute = Omit<PartialSchema, "id"> & {
+  id: PartialAttribute<
+    PartialNumberORM,
+    PartialNumberControlType,
+    number,
+    FinalNumberORM
+  >
+}
+
+export function assembler(schemas: { [schemaName: string]: PartialSchema }): {
+  [schemaName: string]: FinalSchema
+} {
+  return Object.entries(schemas).reduce(
+    (acc, [schemaName, schema]) => ({
+      ...acc,
+      [schemaName]: finalizeSchema(setDefaultPrimaryAttribute(schema)),
```

Comment:
maybe `ensureDefaultPrimaryAttribute` instead of `setDefaultPrimaryAttribute`?

Is there another word that means "make sure this exists if it doesn't"?